File: uptmdigital_app/lib/models/horario.dart
class Horario {
  final int idHorario;
  final int asignaturaId;
  final String dia;
  final String horaInicio;
  final String horaFin;
  final String aula;

  Horario({
    required this.idHorario,
    required this.asignaturaId,
    required this.dia,
    required this.horaInicio,
    required this.horaFin,
    required this.aula,
  });

  factory Horario.fromJson(Map<String, dynamic> json) {
    return Horario(
      idHorario: json['idHorario'],
      asignaturaId: json['asignaturaId'],
      dia: json['dia'],
      horaInicio: json['horaInicio'],
      horaFin: json['horaFin'],
      aula: json['aula'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idHorario': idHorario,
      'asignaturaId': asignaturaId,
      'dia': dia,
      'horaInicio': horaInicio,
      'horaFin': horaFin,
      'aula': aula,
    };
  }
}

File: UPTMDigital.API/Models/Horario.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UPTMDigital.API.Models
{
    public class Horario
    {
        [Key]
        public int IdHorario { get; set; }

        public int AsignaturaId { get; set; }

        [Required]
        public string Dia { get; set; } = string.Empty; // e.g., "Lunes"

        [Required]
        public string HoraInicio { get; set; } = string.Empty; // e.g., "08:00"

        [Required]
        public string HoraFin { get; set; } = string.Empty; // e.g., "10:00"

        public string Aula { get; set; } = string.Empty;

        // Relation (optional if we want to include details)
        // [ForeignKey("AsignaturaId")]
        // public Asignatura? Asignatura { get; set; }
    }
}
